.github/workflows/cluster.yaml

```diff
@@ -0,0 +1,134 @@
+      - uses: actions/checkout@v2
+      - uses: actions/setup-go@v2
+        with:
+          go-version: 1.17.x
+      - name: Install dependencies
+        run: |
+          cd $GITHUB_WORKSPACE
+          go get -u github.com/swaggo/swag/cmd/swag
+          go mod download
+          sudo apt-get update
+          sudo apt-get -y install rpm uidmap
+          # install skopeo
+          . /etc/os-release
+          echo "deb https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable/xUbuntu_${VERSION_ID}/ /" | sudo tee /etc/apt/sources.list.d/devel:kubic:libcontainers:stable.list
+          curl -L https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable/xUbuntu_${VERSION_ID}/Release.key | sudo apt-key add -
+          sudo apt-get update
+          sudo apt-get -y upgrade
+          sudo apt-get -y install skopeo
+          # install notation
+          curl -Lo notation.tar.gz https://github.com/notaryproject/notation/releases/download/v0.7.1-alpha.1/notation_0.7.1-alpha.1_linux_amd64.tar.gz
+          sudo tar xvzf notation.tar.gz -C /usr/bin notation
+          # install oras
+          curl -LO https://github.com/oras-project/oras/releases/download/v0.12.0/oras_0.12.0_linux_amd64.tar.gz
+          mkdir -p oras-install/
+          tar -zxf oras_0.12.0_*.tar.gz -C oras-install/
+          sudo mv oras-install/oras /usr/bin/
+          rm -rf oras_0.12.0_*.tar.gz oras-install/
+
+          # install haproxy
+          sudo apt-get install haproxy
+
+      - name: Run haproxy
+        run: |
+            sudo haproxy -d -f examples/cluster/haproxy.cfg -D
+            sleep 5
+      
+      - name: Prepare configuration files
+        run: |
+            cp test/cluster/config-minio.json test/cluster/config-minio1.json
+            sed -i 's/8081/8082/g' test/cluster/config-minio.json
+            cp test/cluster/config-minio.json test/cluster/config-minio2.json
+            sed -i 's/8082/8083/g' test/cluster/config-minio.json
+            cp test/cluster/config-minio.json test/cluster/config-minio3.json
+
+      - name: Run push-pull tests
+        run: |
+            make binary
+            ./bin/zot-linux-amd64 serve test/cluster/config-minio1.json &
+            ./bin/zot-linux-amd64 serve test/cluster/config-minio2.json &
+            ./bin/zot-linux-amd64 serve test/cluster/config-minio3.json &
+            sleep 10
+            # run without make dependecy "run-minimal", we have cluster
+            make -o run-minimal push-pull
+        env:
+          AWS_ACCESS_KEY_ID: minioadmin
+          AWS_SECRET_ACCESS_KEY: minioadmin
+      
+      - name: Run benchmark
+        run: |
+            ./bin/zot-linux-amd64 serve test/cluster/config-minio1.json &
+            ./bin/zot-linux-amd64 serve test/cluster/config-minio2.json &
+            ./bin/zot-linux-amd64 serve test/cluster/config-minio3.json &
+            sleep 10
+            make -o run-minimal BENCH_OUTPUT=ci-cd run-bench
```

Comment:
Why add a new Makefile target, just build bench and run it directly here.

Reply:
I don't want to run minimal zot, but s3.